clevertap-core/src/test/java/com/clevertap/android/sdk/inapp/customtemplates/TemplatesManagerTest.kt

```diff
@@ -0,0 +1,78 @@
+package com.clevertap.android.sdk.inapp.customtemplates
+
+import com.clevertap.android.sdk.CleverTapInstanceConfig
+import com.clevertap.android.shared.test.BaseTestCase
+import org.junit.Test
+import org.junit.jupiter.api.*
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class TemplatesManagerTest : BaseTestCase() {
+
+    override fun setUp() {
```

Comment:
there should be before and after block ideally, is there a need to use BaseTestCase()?

Reply:
It is extended mainly for having CleverTapInstanceConfig and appCtx provided.

Reply:
I have moved clearRegisteredProducers()  to an @After method.

Reply:
We can mock that class for this test case? are we accessing or operating on it?

Reply:
It can be mocked. I've changed it in the latest commit.